src/components/FriendsPage/FriendsPage.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import ProfileList from './ProfileList';
+import { useRecoilValue } from 'recoil';
+import { friendsInfoSelector } from '../../state/atom';
+import Profile from './Profile';
+import styled from 'styled-components';
+import { AiOutlineSearch } from 'react-icons/ai';
+
+const FriendsPage = () => {
+  const [isSearching, setIsSearching] = useState(false);
+  const [text, setText] = useState('');
+  const friendsInfo = useRecoilValue(friendsInfoSelector);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setText(e.target.value);
+  };
+
+  return (
+    <FriendsPageBox>
+      <Header>
+        <h1>친구</h1>
+        <button onClick={() => setIsSearching(!isSearching)}>
+          <AiOutlineSearch size={30} />
+        </button>
+      </Header>
+
+      {isSearching ? (
+        <>
+          <InputBox>
+            <SearchInput
+              placeholder="친구 검색"
+              value={text}
+              onChange={handleChange}
+            />
+          </InputBox>
+          <>
+            {friendsInfo.map((friend) => (
+              <>
+                {friend.userName.includes(text) ? (
+                  <Profile
+                    key={friend.userId}
+                    userName={friend.userName}
+                    userImg={friend.userImg}
+                    statusMessage={friend.statusMessage}
+                  />
+                ) : (
+                  <></>
+                )}
+              </>
+            ))}
```

Comment:
코드 깊이가 깊어질 수록 가독성이 떨어질 수 있어서 요렇게 return 문을 활용해보시는 것도 좋을 것 같아요!
```suggestion
            {friendsInfo.map((friend) => {
              if(friend.userName.includes(text)){
                return(<Profile
                  key={friend.userId}
                  userName={friend.userName}
                  userImg={friend.userImg}
                  statusMessage={friend.statusMessage}
                />)
              }
            })}
```